rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt

```diff
@@ -1592,10 +1592,17 @@ internal fun TargetContext.toAst(conf: ToAstConfiguration = ToAstConfiguration()
             ReferenceByName(this.getFieldName()),
             toPosition(conf.considerPosition)
         )
-        is IndicatorTargetContext -> IndicatorExpr(
-            this.indic.text.indicatorIndex()!!,
-            toPosition(conf.considerPosition)
-        )
+        is IndicatorTargetContext -> {
+            if (this.indic.text.matches(Regex("\\*IN\\([§£#@\$a-zA-Z]+\\)"))) {
```

Comment:
You are assuming that `*IN(<arg>)` `<arg>` always will be a variable, but is not true.
This statement `*IN(var+1-1)` is accepted, and then you should use the same apprach that I used in the `IndicatorContext.toAst`

Add some test in `INDIC01.rpgle`

Reply:
I agree with you. Focused to the task, I didn't think the case with an internal expression. In this case, I'll change `RpgLexer.g4` to consider other cases, then the logic `is IndicatorTargetContext` like you suggested.